app/javascript/controllers/calendar_controller.js

```diff
@@ -0,0 +1,28 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["date"]
+
+  connect() {
+    this.dateTargets.forEach((date) => {
+      date.parentElement.addEventListener("click", this.toggleSelect)
```

Comment:
Generally, if you add an event listener somewhere, you need to also remove it when the stimulus controller disconnects.

Might be better to move this into the parent elements `data-action` instead?  Then stimulus handles all the event listening/disconnecting for you.